src/test/java/com/wikia/webdriver/common/core/url/UrlBuilder.java

```diff
@@ -50,90 +76,105 @@ public String getUrlForPath(String wikiPath) {
-      return url + separator + qs;
+  public String getUrlForWiki(String wikiName, boolean addWWW) {
+    final String wikiaName = getWikiaGlobalName(wikiName);
+    EnvType envType = Configuration.getEnvType(this.env);
+
+    String www = "";
+    if (addWWW) {
+      www = "www.";
     }
-  }
 
-  private String composeUrl(String prefix, String wikiName, String suffix) {
-    String overwrittenWikiName = wikiName;
-    String overwrittenPrefix = prefix;
-    if (env != null) {
-      if ((!env.contains("dev") || env.contains(SANDBOX_MERCURY_DEV)) && !"prod".equals(env)) {
-        overwrittenPrefix = env + "." + prefix;
+    switch (envType) {
+      case DEV: {
+        String devBoxOwner = this.env.split("-")[1];
+        return String.format(getUrlProtocol() + DEV_URL_FORMAT, www, wikiaName, devBoxOwner, envType.getWikiaDomain());
       }
-
-      if (env.contains("dev") && !env.contains(SANDBOX_MERCURY_DEV) &&
-          !"CHROMEMOBILEMERCURY".equalsIgnoreCase(browser) && wikiName.endsWith("wikia")) {
-        overwrittenWikiName = "wikiaglobal";
+      case PROD: {
+        return String.format(getUrlProtocol() + PROD_URL_FORMAT, www, wikiaName, envType.getWikiaDomain());
+      }
+      case STAGING: {
+        return String.format(getUrlProtocol() + PROD_URL_FORMAT, www, wikiaName, envType.getWikiaDomain());
       }
+      case SANDBOX: {
+        if (newStagingUrlFormat) {
+          return String.format(getUrlProtocol() + SANDBOX_NEW_URL_FORMAT, www, wikiaName, this.env, envType.getWikiaDomain());
+        } else {
+          return String.format(getUrlProtocol() + SANDBOX_URL_FORMAT, this.env, www, wikiaName, envType.getWikiaDomain());
+        }
+      }
+      default:
+        return "";
     }
-
-    return "http://" + overwrittenPrefix + overwrittenWikiName + suffix + "/";
   }
 
-  public static String getUrlForPageWithWWW(String pageName) {
-    String environment = Configuration.getEnv();
-    String wikiName = Configuration.getWikiName();
-
-    environment = environment.equals("prod") ? "" : environment + ".";
-    wikiName = wikiName.equals("wikia") ? "" : wikiName + ".";
-
-    return environment + "www." + wikiName + "wikia.com" + pageName;
+  private String getUrlProtocol() {
+    if (this.forceHttps) {
+      return "https://";
+    }
+    return "http://";
   }
 
-  public static String getHostForWiki() {
-    String environment = Configuration.getEnv();
-    String wikiName = Configuration.getWikiName();
+  public String getWikiGlobalURL(){
+    EnvType env = Configuration.getEnvType(this.env);
 
-    environment = environment.equals("prod") ? "" : environment + ".";
-    wikiName = wikiName.equals("wikia") ? "www." : wikiName + ".";
-
-    return environment + wikiName + "wikia.com";
+    switch (env) {
+      case DEV: {
+        String devBoxOwner = this.env.split("-")[1];
+        return String.format(getUrlProtocol() + "www.%s.%s", devBoxOwner, env.getWikiaDomain());
+      }
+      case SANDBOX: {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract this nested code block into a method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1199)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this switch case number of lines from 7 to at most 5, for example by extracting code into methods. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1151)